### ISSUE TITLE
Feature Request: Food Spoilage

### ISSUE THREAD

**Issue by I3ladeDragon** [MLG-Fortress/ExtraHardMode#202]
Basically allow inventory and chest foods to spoil over a period of time.

Example: 
- Raw Chicken turns into Rotten Flesh after 6 IRL hours.
- Cooked Chicken turns into Rotten Flesh after 72 IRL hours.

**Comment by ZelenePeklo** [comment]
I like this. Is it feasible?

**Comment by RoboMWM** [comment]
Not really feasible unless we monitor chests. Depending on implementation, this may cause chunk loads, which are crazy expensive in 1.16 given the issues I've been receiving on chunk load operations. Lastly, this could kinda be a feature left to extra expansion to really make good use. E.g., how about putting ice to preserve food, and make it melt after some time?

Patrons (which I may make a tier for, as I think about this) may request features to be of higher priority.

**Comment by RoboMWM** [comment]
Either way, biggest issue is this introduces a new concept (time-based actions) that would have to be defined (does time occur only while player is logged in? Chunk loaded? All the time?) and then explained to players, so this would take significant work to create.

**Comment by I3ladeDragon** [comment]
I didn't really think deeply about how this would be implemented when I made this suggestion, but I do like the idea of expanding upon the idea to use ice to help preserve food!

Since you cannot keep every chest loaded without destroying performance, it should probably only run inside of already loaded chunks and stop when the chunk is no longer loaded.

Nifty idea, for sure, but I can see where implementation could be a different story.

**Comment by RoboMWM** [comment]
> I think if you wanted it to constantly run, even for chests in unloaded chunks, you would need SQL database support

Yes, would need persistence (doesn't have to be SQL but that is one form of persistence) and would also need a way to ID those individual items.

But using gherkin statements would help break up this feature into implementable chunks if you want to pursue. https://content.pivotal.io/blog/how-to-write-well-formed-user-stories Idk how much time I'll have but this project does attract PRs time to time.
